controllers/mdutil/util.go

```diff
@@ -697,3 +709,15 @@ func ComputeHash(template *clusterv1.MachineTemplateSpec) uint32 {
 	DeepHashObject(machineTemplateSpecHasher, *template)
 	return machineTemplateSpecHasher.Sum32()
 }
+
+// GetDeletingMachineCount gets the number of machines that are in the process of being deleted
+// in a machineList.
+func GetDeletingMachineCount(machineList *clusterv1.MachineList) int32 {
+	var deletingMachineCount int32 = 0
+	for _, machine := range machineList.Items {
+		if machine.DeletionTimestamp != nil {
```

Comment:
nit: technically  a deletionTimestamp  could still be zero.
Could use `if !machine.GetDeletionTimestamp().IsZero() {}`

Reply:
addressed